test/parallel/test-fs-long-path.js

```diff
@@ -1,34 +1,46 @@
 'use strict';
-var common = require('../common');
-var fs = require('fs');
-var path = require('path');
-var assert = require('assert');
+const common = require('../common');
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
 
-var successes = 0;
+// when it fails test again under real OS tmp dir on Linux when it is
+// readable/writable to avoid failing tests on ecryptfs filesystems:
+// https://github.com/nodejs/node/issues/2255
+// it follows advice in comments to:
+// https://github.com/nodejs/node/pull/3925
+// https://github.com/nodejs/node/pull/3929
+try {
+  common.refreshTmpDir();
+  testFsLongPath(common.tmpDir);
+  common.refreshTmpDir();
+} catch (e) {
+  if (os.type() == 'Linux') {
```

Comment:
@jbergstroem Yes, it would be the same. This code already uses the `os` module for other things so it seemed to make sense to use the normalized name for the comparison but it could test `process.platform` or `os.platform()` just as well as `os.type()`. I can change it if that's what's holding the PR #3936 back.